src/main/java/com/xjeffrose/chicago/ChicagoMessageAggregator.java

```diff
@@ -0,0 +1,37 @@
+package com.xjeffrose.chicago;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageDecoder;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ChicagoMessageAggregator extends MessageToMessageDecoder<ChicagoMessage> {
+  private final AtomicBoolean started = new AtomicBoolean();
+  private AggregatedChicagoMessage aggregatedMessage;
+
+  private void aggregate(ChicagoMessage msg) {
+    if (started.get()) {
+      aggregatedMessage.appendVal(msg.getVal());
+    } else {
+      aggregatedMessage = new AggregatedChicagoMessage(msg);
+      started.set(true);
+    }
+  }
+
+
+  @Override
+  protected void decode(ChannelHandlerContext ctx, ChicagoMessage msg, List<Object> out) throws Exception {
+
+    switch (msg.getOp()) {
+      case STREAM_RESPONSE:
+        aggregate(msg);
+        if (new String(msg.getVal()).contains(ChiUtil.delimiter)) {
+          out.add(aggregatedMessage);
```

Comment:
I see you changed this but want to understand how it will work.

Won't the pipeline/channel get re-used in the RequestMuxer connection pool?  That would mean all the handlers should expect to see multiple messages and potentially multiple stream responses, right?